### DISCUSSION TITLE
Issue loading video file using Git LFS

### DISCUSSION THREAD

**Discussion by jenkatan29** [community/community#52612] [New to GitHub] [answered]
Hello, 

I'm adding a 36MB file to my website but repeatedly getting an error saying that the file size is larger than githubs recommended maxmum. The size in the error does not match the size on the video properties or using the du command. I added it using Git LFS but still getting the same request to readd using Git LFS. 

`$ du -sh profileVideo_Jen.mp4
36M     profileVideo_Jen.mp4
`
Expected result: 
Since it is less than 50MB, expected to upload without error. 
 
Actual result: 
Unable to upload and getting an error each time. 
`$ git push origin main
Uploading LFS objects: 100% (6/6), 97 MB | 0 B/s, done.
Enumerating objects: 69, done.
Counting objects: 100% (69/69), done.
Delta compression using up to 8 threads
Compressing objects: 100% (46/46), done.
Writing objects: 100% (47/47), 300.21 MiB | 9.28 MiB/s, done.
Total 47 (delta 25), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (25/25), completed with 14 local objects.
remote: warning: File images/profileVideo_Jen.mp4 is 72.50 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
remote: error: Trace: 226ff6d9c9d56c4e13c87aa8c11592662d3560eac2d5efcfb9d7b4ab0b50ce66
remote: error: See https://gh.io/lfs for more information.
remote: error: File images/profileVideo_Jen.mp4 is 193.19 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
To https://github.com/jenkatan29/jenkatan29.github.io.git
 ! [remote rejected] main -> main (pre-receive hook declined)
error: failed to push some refs to 'https://github.com/jenkatan29/jenkatan29.github.io.git'`

Any advice? Thanks for all your help 
`

**Comment by airtower-luna** [comment, marked as answer]
You'll have to look at the repository history to solve this. Most likely you committed a larger version of the file at some point in the past, so it's still present in the old commit. When you push your current commit all its ancestors need to be pushed too.

If my guess is correct, you'll have to [rewrite the repository history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) to remove the large files (or move them to LFS) before you can push your repository to GitHub. See also: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository

**Reply by jenkatan29** [reply]
The bfg cleaner worked perfectly, thanks for the tip!

**Reply by jenkatan29** [reply]
I was able to make all my changes but it looks like all of the videos on my site are not working, even though I only targeted videos over 100M and all the other videos on my site were much smaller. All the file paths have not changed. The only thing that's changed is that they are being tracked by Git LFS. Any idea what could be the issue? I can start a new discussion as well since this is a bit of a tangent.

**Reply by airtower-luna** [reply]
I'd suspect that the Pages deployment doesn't fetch files from LFS. But if you want to dig deeper into that combination a separate topic is probably best. :sweat_smile:

**Reply by jenkatan29** [reply]
thanks for the tip!

**Reply by gsheridan-pgi** [reply]
Hi jenkatan29, I know this is from a couple of years ago, but do you remember how you resolved the problem of not being able to play videos on your site after you started tracking them with Git LFS. I'm having the same problem.
